### ISSUE TITLE
Populate Hosted Weblate translation project

### ISSUE THREAD

**Issue by comradekingu** [strongbox-password-safe/Strongbox#406]
https://hosted.weblate.org/projects/strongbox/#information is up, but not populated.
I can help if you add https://hosted.weblate.org/user/kingu/ as an admin. :)

**Comment by strongbox-mark** [comment]
Thank you, yes, would appreciate the help. I've added your there as an admin.

**Comment by comradekingu** [comment]
@strongbox-mark Done, you have to set up webhooks, because I can't do that on here.
https://docs.weblate.org/no/latest/admin/continuous.html#automatically-receiving-changes-from-github
It is possible to write your own script too, but that is a bit of a faff.

**Comment by strongbox-mark** [comment]
Hi @comradekingu - Sorry for the late response. Yes, existing translations are in a different repo here:

https://github.com/strongbox-password-safe/babel

Long story but they must be in a separate repository for licensing reasons... I've tried to fix this up now, but I'm not sure I've done it correctly...

Also, there are 3 directories that contain strings files in that repository. All of these directories and files need to be translated/added, but from what I can see only Localizable.strings is?

Thanks! I can setup the webhooks, no problem

**Comment by comradekingu** [comment]
@strongbox-mark I think the filemask fails for https://github.com/strongbox-password-safe/babel/tree/master/StrongBox, as it is set up to be StrongBox/*.lproj/Localizable.strings in https://hosted.weblate.org/settings/strongbox/translations/#files

**Comment by strongbox-mark** [comment]
Ok, but I don’t know what it should be? Any suggestions?

*/*.lproj/*.strings doesn’t seem to work

**Comment by comradekingu** [comment]
StrongBox/*/Localizable.strings I would imagine for filemask, but don't know what base file should be.
Tried ../Localizable.strings and Strongbox/Localizable.strings
Maybe it is confused by originally being pointed to another repo?

**Comment by strongbox-mark** [comment]
yeah, I mean I don't know what's wrong at all. I've been trying to set this up for a while but I've had to give up. If you know anyone involved in Weblate that could help that would be great, but I can't commit much more time to it right now.

**Comment by comradekingu** [comment]
I made a new one and it is importing the strings with the settings it wouldn't accept when changing the repo over.

**Comment by strongbox-mark** [comment]
Awesome!!

**Comment by comradekingu** [comment]
:) Set up the hooks and that should work. I'll put in a bugreport over at Weblate.

**Comment by strongbox-mark** [comment]
Will do... it seems like only one file is available? Localizable.strings. Isn’t it possible to add all the others?

**Comment by comradekingu** [comment]
https://github.com/WeblateOrg/weblate/issues/4535
Basically https://hosted.weblate.org/projects/strongbox/translations/ can't be saved (?)
and
https://hosted.weblate.org/projects/strongbox/strongbox/ seems to work

Don't delete the "translations" one though, as it may be useful for debugging why that is.
Freeze the Crowdin integration, and set up the webhooks to https://hosted.weblate.org/projects/strongbox/translations/

Curiously https://hosted.weblate.org/projects/strongbox/ is "Strongbox" in the name "and "babel" is small b in https://github.com/strongbox-password-safe ;)

What is https://github.com/strongbox-password-safe/babel/tree/master/StrongBox%20Auto%20Fill btw?

**Comment by strongbox-mark** [comment]
No worries. Won't delete it.

Re: CrowdIn - This will stay as primary platform until Weblate is shown to work. CrowdIn has worked well for all languages so far. That's not something to be sniffed at... The two platforms will be orthogonal, both commit to different branches and I will manage the merge to master, and the copy into the final build process.

The "Strongbox Auto Fill" component is the set of strings used in the App Extension for the iOS "AutoFill" subsystem. This allows users to tap on a login in Safari or other App, and have their passwords filled in automatically.

**Comment by strongbox-mark** [comment]
Hi @comradekingu - Any luck adding those other strings files? No rush, just curious.

**Comment by comradekingu** [comment]
@strongbox-mark Webhooks are not set up, and I can't do it because I am not an admin on here.
I deleted the non-working component. Applied to have the libre hosting approved.
What are other *.strings for? Where are they?

**Comment by strongbox-mark** [comment]
The *.strings are for all the different UI components. 

e.g. https://github.com/strongbox-password-safe/babel/blob/master/StrongBox/nb/Audit.strings

Localizable.strings is the largest file and it is the file used directly in code, but all Interface Builder (GUI designed components) have individual strings files. It's just how iOS/Mac development does things.

All of those strings files need to be translated.

I'll try to get webhooks setup today...

**Comment by comradekingu** [comment]
I can't get it to accept what I am trying to do, but I can make new components for each file.

**Comment by strongbox-mark** [comment]
Weird, I'm not sure how it should be setup. Are there downsides to doing it as separate components?

**Comment by comradekingu** [comment]
It is just more work. I found a plugin that does component discovery, I think that adds stuff automatically, but it doesn't accept a file level wildcard as the basis of new translations.

**Comment by comradekingu** [comment]
@strongbox-mark It works now, turns out it had to be {{ component }} for the base file field. Big thanks to @orangesunny :)))
https://hosted.weblate.org/projects/strongbox/ (deleted the old component)
Adding new files should add components automatically now. Not sure about message for translators of all the other jazz, but add the webhook and it should be good to go.

**Comment by strongbox-mark** [comment]
That sounds great! Nice work.

I believe I've now added the webhooks too!

**Comment by comradekingu** [comment]
@strongbox-mark Can you screenshot the "Webhooks" settings?

> ERROR: Permission to strongbox-password-safe/babel.git denied to weblate.
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
>  (128)

I have no clue why _reading_ from the remote repo would fail.
Edit: Well it was actually as stupid as it would seem, no push address set up.
Edit2: No, all the components share the info from the "strings" component, and that is set up correctly

https://hosted.weblate.org/changes/?page=2&limit=20&component=strings&project=strongbox
Looks to have worked just fine from you set it up on the GitHub end, till two days ago.
But that isn't the case, because nothing has landed in
https://github.com/strongbox-password-safe/babel/commits/weblate

I don't know, still think the settings on GitHub are iffy.

**Comment by strongbox-mark** [comment]
I've added/invited you now to the babel project @comradekingu - Please feel free to change as you feel appropriate there... As long as you don't commit/push to master that is. 

I will manage merging localized branches to master - once they are ready. 

The webhook settings currently look like this:

![image](https://user-images.githubusercontent.com/64356496/94994510-ef688a00-058f-11eb-926a-67c2e3dcf6e8.png)

**Comment by strongbox-mark** [comment]
I would like now to move this issue over to the babel repository as it is entirely related to the localization process...
